components/LoginForm.vue

```diff
@@ -1,59 +1,92 @@
-    <v-text-field
-      v-model="login.username"
-      outlined
-      label="Username"
-    ></v-text-field>
-    <v-text-field
-      v-model="login.pw"
-      outlined
-      label="Password"
-      :append-icon="showPassword ? 'mdi-eye-off' : 'mdi-eye'"
-      :type="showPassword ? 'text' : 'password'"
-      @click:append="() => (showPassword = !showPassword)"
-    ></v-text-field>
-    <v-btn block large color="primary" type="submit">submit</v-btn>
-  </v-form>
+  <div>
+    <v-form ref="form" v-model="validInputs" @submit.prevent="userLogin">
+      <v-text-field
+        v-model="serverUrl"
+        outlined
+        label="Server Address"
+        type="url"
+        :rules="[(v) => !!v || 'Server address is required']"
+        required
+      ></v-text-field>
+      <v-text-field
+        v-model="login.username"
+        outlined
+        label="Username"
+        :rules="[(v) => !!v || 'Username is required']"
+        required
+      ></v-text-field>
+      <v-text-field
+        v-model="login.pw"
+        outlined
+        label="Password"
+        :append-icon="showPassword ? 'mdi-eye-off' : 'mdi-eye'"
+        :type="showPassword ? 'text' : 'password'"
+        @click:append="() => (showPassword = !showPassword)"
+      ></v-text-field>
+      <v-btn :disabled="!validInputs" block large color="primary" type="submit"
+        >submit</v-btn
+      >
+    </v-form>
+    <v-alert v-if="errorMessage" class="mt-5" outlined type="error"
+      >{{ errorMessage }}
+    </v-alert>
+  </div>
 </template>
 
 <script lang="ts">
 export default {
   data() {
     return {
-      serverUrl: 'http://localhost:8096',
+      serverUrl: '',
       login: {
         username: '',
         password: ''
       },
-      showPassword: false
+      showPassword: false,
+      errorMessage: '',
+      validInputs: false
     };
   },
   methods: {
     async userLogin() {
-      try {
-        this.$axios.setBaseURL(this.serverUrl);
-        const response = await this.$auth.loginWith('local', {
-          data: this.login
-        });
-        this.$auth.setUserToken(
-          // TODO: Generate the token properly
-          `MediaBrowser Client="Jellyfin Web", Device="Firefox", DeviceId="TW96aWxsYS81LjAgKFgxMTsgTGludXggeDg2XzY0OyBydjo3Ny4wKSBHZWNrby8yMDEwMDEwMSBGaXJlZm94Lzc3LjB8MTU5NTQ1MTYzMzE4OQ11", Version="10.7.0", Token="${response.data.AccessToken}"`
-        );
-        this.$axios.setToken(
-          // TODO: Generate the token properly
-          `MediaBrowser Client="Jellyfin Web", Device="Firefox", DeviceId="TW96aWxsYS81LjAgKFgxMTsgTGludXggeDg2XzY0OyBydjo3Ny4wKSBHZWNrby8yMDEwMDEwMSBGaXJlZm94Lzc3LjB8MTU5NTQ1MTYzMzE4OQ11", Version="10.7.0", Token="${response.data.AccessToken}"`,
-          'X-Emby-Authorization'
-        );
-        this.$auth.setUser(response.data.User);
-      } catch (err) {
-        console.error('Failed to login:', err);
+      const testHttp = /^https?:\/\//;
+
+      if (!testHttp.test(this.serverUrl)) {
```

Comment:
You should define a rule in the data like this:

```javascript
serverAddressRules: [ 
    v => !!v || 'Server address is required', 
    v => /^https?:\/\//.test(v) || 'Server address must be a valid URL' 
]
```
(Very untested)

Then use it like:
```javascript
:rules="serverAddressRules"
```

Reply:
Yeah that works, I've tested, and updated it 👍